soes/hal/rt-kernel-lan9252/esc_hw.c

```diff
@@ -91,32 +91,23 @@ static void lan9252_write_32 (uint16_t address, uint32_t val)
     data[5] = ((val >> 16) & 0xFF);
     data[6] = ((val >> 24) & 0xFF);
 
-    /* Select device. */
-    spi_select (lan9252);
     /* Write data */
     write (lan9252, data, sizeof(data));
-    /* Un-select device. */
-    spi_unselect (lan9252);
 }
 
 /* lan9252 single read */
 static uint32_t lan9252_read_32 (uint32_t address)
 {
-   uint8_t data[4];
+   uint8_t data[2];
    uint8_t result[4];
+   uint16_t lseek_addr;
 
-   data[0] = ESC_CMD_FAST_READ;
-   data[1] = ((address >> 8) & 0xFF);
-   data[2] = (address & 0xFF);
-   data[3] = ESC_CMD_FAST_READ_DUMMY;
+   data[0] = ((address >>8) & 0xFF);
+   data[1] = (address & 0xFF);
 
-   /* Select device. */
-   spi_select (lan9252);
-   /* Read data */
-   write (lan9252, data, sizeof(data));
+	lseek_addr=((uint16_t)data[0] << 8) | data[1];
+	lseek (lan9252, lseek_addr, SEEK_SET);
```

Comment:
Replace lseek() with write() function when register is read.
![2](https://user-images.githubusercontent.com/10510127/28055161-bc660202-6652-11e7-8583-69a8f571a582.png)